torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -281,6 +283,21 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(guard_scalar(val))
                     graph.erase_node(node)
 
+    should_restart = False
+    for i, node in enumerate(graph.nodes):
+        if node.op == "placeholder" and len(node.users) == 0:
+            # At this point we've lost the back pointer to
+            # what f_local this placeholder points to. Instead,
+            # we will rely on the index to specialize when we
+            # restart analysis.
```

Comment:
This is probably the most controversial part of this PR. An alternative universe is to have AOTAutoGrad propogate placeholder names to this point and we can have hoist the specialization logic up to the wrap_symfloat level.